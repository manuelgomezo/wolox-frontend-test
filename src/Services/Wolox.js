import axios from 'axios';
import config from 'config';

const woloxService = {
  async login(email, password, signal) {
    const response = await axios
      .post(`${config.WOLOX_API}/login`, { email, password }, { ...(signal && { cancelToken: signal }) })
      .catch((error) => error);

    if (response.status === 200 && response.data) {
      return [response.data, null];
    }
    return [null, response];
  },

  async getList(name, signal) {
    const response = await axios.get(`${config.WOLOX_API}/${name}`, { ...(signal && { cancelToken: signal }) }).catch((error) => error);

    if (response.status === 200 && response.data) {
      return [response.data, null];
    }
    return [null, response];
  },
};

export default woloxService;
