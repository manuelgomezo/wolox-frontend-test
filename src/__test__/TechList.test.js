import React from 'react';
import { render, screen } from '@testing-library/react';
import { create, act } from 'react-test-renderer';

import { TechList } from 'Pages';
import Wolox from 'Services/Wolox';

// jest.mock('react', () => ({
//   ...jest.requireActual('react'),
// }));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

describe('TechList getList correct response', () => {
  beforeAll(() => {
    Wolox.getList = jest.fn(() => {
      return new Promise((res) => {
        const testObject = {
          tech: 'Manuel Testing',
        };
        res([[testObject], null]);
      });
    });
  });

  it('should render TechList correct component', () => {
    render(<TechList />);
    // const card = screen.getByText(/Manuel Testing/i);
  });
});

describe('TechList getList error response', () => {
  beforeAll(() => {
    Wolox.getList = jest.fn(() => {
      return new Promise((res) => {
        res([null, new Error('Testing error')]);
      });
    });
  });

  it('should render TechList error component', () => {
    let root;
    act(() => {
      root = create(<TechList />);
    });
    expect(root).toThrow(TypeError);
  });
});
