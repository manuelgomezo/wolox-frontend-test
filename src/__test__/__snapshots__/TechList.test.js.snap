// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechList getList error response should render TechList error component 1`] = `
<div
  className="tech-list max-width"
>
  <div
    className="tech-list__filters"
  >
    <div
      className="filter-content"
    >
      <div
        className="filter-content__item"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="filter.search"
          value=""
        />
      </div>
      <div
        className="filter-content__item"
      >
        <select
          className="input"
          defaultValue="default"
          onChange={[Function]}
        >
          <option
            disabled={true}
            value="default"
          >
            filter.sort
          </option>
          <option
            value="desc"
          >
            filter.abc
          </option>
          <option
            value="asc"
          >
            filter.cba
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="tech-list__container"
  >
    <div
      className="loading"
    />
  </div>
</div>
`;
